pkg/acsengine/engine.go

```diff
@@ -506,8 +506,12 @@ func getParameters(cs *api.ContainerService, isClassicMode bool, generatorCode s
 			addSecret(parametersMap, "etcdServerPrivateKey", properties.CertificateProfile.EtcdServerPrivateKey, true)
 			addSecret(parametersMap, "etcdClientCertificate", properties.CertificateProfile.EtcdClientCertificate, true)
 			addSecret(parametersMap, "etcdClientPrivateKey", properties.CertificateProfile.EtcdClientPrivateKey, true)
-			addArraySecret(parametersMap, "etcdPeerCertificates", properties.CertificateProfile.EtcdPeerCertificates, true)
-			addArraySecret(parametersMap, "etcdPeerPrivateKeys", properties.CertificateProfile.EtcdPeerPrivateKeys, true)
+			for i, pc := range properties.CertificateProfile.EtcdPeerCertificates {
```

Comment:
Since this parameter is only relevant for clusters with 1+ masters, would this need some kind of "!HostedMaster" guard?

Reply:
how about `properties.MasterProfile != nil`?